src/main/java/com/minelittlepony/render/layer/LayerPonyArmor.java

```diff
@@ -118,113 +122,48 @@ private void renderArmor(T entity, float move, float swing, float partialTicks,
         }
     }
 
-    @SuppressWarnings("incomplete-switch")
     private void prepareToRender(ModelPonyArmor model, EntityEquipmentSlot slot, boolean isPony) {
         model.setVisible(false);
 
         switch (slot) {
-            // feet
+            case HEAD:
+                model.showHead(isPony);
+                break;
             case FEET:
-                model.bipedRightArm.showModel = true;
-                model.bipedLeftArm.showModel = true;
-                model.bipedRightLeg.showModel = !isPony;
-                model.bipedLeftLeg.showModel = !isPony;
-                model.leftLegging.showModel = isPony;
-                model.rightLegging.showModel = isPony;
+                model.showFeet(true);
                 break;
-            // legs
             case LEGS:
-                model.bipedRightLeg.showModel = !isPony;
-                model.bipedLeftLeg.showModel = !isPony;
-                model.bipedRightArm.showModel = true;
-                model.bipedLeftArm.showModel = true;
-                model.bipedBody.showModel = !isPony;
-                model.flankGuard.showModel = !isPony;
-                model.saddle.showModel = isPony;
-                model.leftLegging.showModel = isPony;
-                model.rightLegging.showModel = isPony;
-                break;
-            // chest
+                model.showFeet(true);
+                model.showLegs(isPony);
             case CHEST:
-                model.saddle.showModel = isPony;
-                model.bipedBody.showModel = !isPony;
-                model.flankGuard.showModel = !isPony;
-                break;
-            // head
-            case HEAD:
```

Comment:
@killjoy1221 
I rearranged them to reduce duplication. So the case for CHEST will fall through because it is a superset of the case for HEAD.

Reply:
@killjoy Correction: _LEGS_ fall through to _CHEST_ because the latter shows only the saddle whilst the former shows the saddle _and_ the legs. This is one of those rare cases where switch statement's ability to stack conditions is actually useful.

Reply:
I think you meant @killjoy1221.

Reply:
@killjoy Oh, sorry! I didn't realise there were more than one @killjoy1221